packages/zoe/src/zoe.js

```diff
@@ -323,56 +323,56 @@ const makeZoe = (additionalEndowments = {}) => {
         return harden(redemptionResult);
       };
 
-      const inviteAmount = inviteIssuer.burn(invite);
-      assert(
-        inviteAmount.extent.length === 1,
-        'only one invite should be redeemed',
-      );
+      return inviteIssuer.burn(invite).then(inviteAmount => {
```

Comment:
I think this is just my personal preference, but I like taking any success callback and rather than having an anonymous function as we have here, naming it above and then using the name. To me, it's easier to look at the flow of multiple named functions rather than anonymous callbacks.

Reply:
I feel the same way. 

I pulled out two `.then` bodies, but I didn't see any improvement from extracting the `.then` of the `burn()` call, so I put it back.  It looks more like "do everything else", and it's the final step. 

I like the way its final statement is 
```
return Promise.all(...)
  .then(recordOfferAndPayout)
  .then(makeRedemptionResults)
```
That structure was harder to see when I extracted the method.